src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -206,22 +250,30 @@ Namespace Microsoft.CodeAnalysis.Semantics
         Private Function CreateBoundAssignmentOperatorOperation(boundAssignmentOperator As BoundAssignmentOperator) As IOperation
             Dim kind = GetAssignmentKind(boundAssignmentOperator)
             If kind = OperationKind.CompoundAssignmentExpression Then
-                Dim binaryOperationKind As BinaryOperationKind = DirectCast(Create(boundAssignmentOperator.Right), IBinaryOperatorExpression).BinaryOperationKind
+                ' convert Right to IOperation temporarily. we do this to get right operand, operator method and etc
+                Dim temporaryRight = DirectCast(Create(boundAssignmentOperator.Right), IBinaryOperatorExpression)
+
+                Dim binaryOperationKind As BinaryOperationKind = temporaryRight.BinaryOperationKind
                 Dim target As Lazy(Of IOperation) = New Lazy(Of IOperation)(Function() Create(boundAssignmentOperator.Left))
-                Dim value As Lazy(Of IOperation) = New Lazy(Of IOperation)(Function() DirectCast(Create(boundAssignmentOperator.Right), IBinaryOperatorExpression).RightOperand)
-                Dim usesOperatorMethod As Boolean = DirectCast(Create(boundAssignmentOperator.Right), IBinaryOperatorExpression).UsesOperatorMethod
-                Dim operatorMethod As IMethodSymbol = DirectCast(Create(boundAssignmentOperator.Right), IBinaryOperatorExpression).OperatorMethod
+
+                ' right now, parent of right operand is set to the temporary IOperation, reset the parent
+                ' we basically need to do this since we skip BoundAssignmentOperator.Right from IOperation tree
+                Dim rightOperand = Operation.ResetParentOperation(temporaryRight.RightOperand)
```

Comment:
Can another thread ask for `rightOperand.Parent` before it's reset? What are the implications?